plugins/oauth/girder_oauth/__init__.py

```diff
@@ -17,15 +17,28 @@ def checkOauthUser(event):
     if user.get('oauth'):
         if isinstance(user['oauth'], dict):
             # Handle a legacy format where only 1 provider (Google) was stored
-            prettyProviderNames = 'Google'
+            userProviders = ['google']
         else:
+            userProviders = [val['provider'] for val in user['oauth']]
+
+        # Silently skip non-installed providers
+        availableUserProviders = [
+            userProvider for userProvider in userProviders
+            if userProvider in providers.idMap
+        ]
```

Comment:
```python
    userProviders = {'google'}
else:
    userProviders = {val['provider'] for val in user['oauth']}

availableUserProviders = set(providers.idMap.keys()) & userProviders
```

What do you think? Feels like the order doesn't matter here.

Reply:
I was operating as though ordering mattered when I wrote this, but you're right, it doesn't. Set comparison is more natural in this case.

Reply:
Changed.